addons/mass_mailing/tests/test_mailing_ui.py

```diff
@@ -79,3 +79,27 @@ def test_snippets_mailing_menu_toolbar_mobile_tour(self):
 
     def test_mass_mailing_code_view_tour(self):
         self.start_tour("/web?debug=tests", 'mass_mailing_code_view_tour', login="demo")
+
+
+class TestMassMailingActions(MassMailCommon):
+    def test_mailing_action_open(self):
+        mass_mailings = self.env['mailing.mailing'].create([
+            {'subject': 'First subject'},
+            {'subject': 'Second subject'}
+        ])
+        # Create two traces: one linked to the created mass.mailing and one not (action should open only the first)
+        self.env["mailing.trace"].create([{
+                "trace_status": "open",
+                "mass_mailing_id": mass_mailings[0].id,
+                "model": "res.partner",
+                "res_id": self.partner_admin.id,
+            }, {
+                "trace_status": "open",
+                "mass_mailing_id": mass_mailings[1].id,
+                "model": "res.partner",
+                "res_id": self.partner_employee.id,
+            }
+        ])
+        results = mass_mailings[0].action_view_opened()
+        results_partners = self.env["res.partner"].search(results['domain'])
+        self.assertEqual(len(results_partners), 1, "Trace leaked from mass_mailing_2 to mass_mailing_1")
```

Comment:
Could check it is partner_admin directly probably ?

Reply:
yup, that works as well